pro/src/screens/IndividualOffer/SummaryScreen/SummaryScreen.tsx

```diff
@@ -94,26 +77,25 @@ const SummaryScreen = () => {
     }
 
     setIsDisabled(true)
-    const venueHasCreatedNonFreeOffer = isNewBankDetailsJourneyEnabled
-      ? venueList.some((venue) => {
-          return venue.hasNonFreeOffers
-        })
-      : undefined
-
+    const offererResponse = isNewBankDetailsJourneyEnabled
+      ? await api.getOfferer(offer.offererId)
+      : null
     const publishIndividualOfferResponse = await publishIndividualOffer({
       offerId: offer.id,
     })
+
     if (publishIndividualOfferResponse.isOk) {
       setOffer?.(serializeOfferApi(publishIndividualOfferResponse.payload))
+      const shouldDisplayRedirectDialog =
+        (isNewBankDetailsJourneyEnabled &&
+          publishIndividualOfferResponse.payload.isNonFreeOffer &&
+          offererResponse &&
+          !offererResponse.hasNonFreeOffer &&
+          !offererResponse.hasValidBankAccount &&
+          !offererResponse.hasPendingBankAccount) ||
+        (!isNewBankDetailsJourneyEnabled && showVenuePopin[venueId || ''])
```

Comment:
C'est peut-être plus compréhensible de rajouter des `const` pour expliciter la condition.

```suggestion
    const offererWithoutNonFreeOffers =
        offererResponse &&
          !offererResponse.hasNonFreeOffer 
     const offererWithoutValidOrPendingBankAccount =
        offererResponse &&
          !offererResponse.hasValidBankAccount &&
          !offererResponse.hasPendingBankAccount
      const shouldDisplayRedirectDialog =
        (isNewBankDetailsJourneyEnabled &&
          publishIndividualOfferResponse.payload.isNonFreeOffer &&
          offererWithoutNonFreeOffers && 
          offererWithoutValidOrPendingBankAccount) ||
        (!isNewBankDetailsJourneyEnabled && showVenuePopin[venueId || ''])
```

Reply:
Hmmmm j'ai pas l'impression que c'est beaucoup plus lisible, notamment le 
```
const offererWithoutNonFreeOffers =
        offererResponse &&
          !offererResponse.hasNonFreeOffer
```

Au final je trouve le `!offererResponse.hasNonFreeOffer` tout aussi lisible, mais j'imagine que c'est une question de goût